opencga-app/app/scripts/docker/Makefile.docker

```diff
@@ -0,0 +1,95 @@
+#!make
+
+# Override the `make_env` file default location by setting ENVFILE env var
+ifneq ("$(wildcard $(ENVFILE))","")
+$(info Makefile.docker: including environment file $(ENVFILE))
+include $(ENVFILE)
+export
+endif
+
+# Check required variables are set for target build
+ifeq ($(MAKECMDGOALS), build)
+ifeq ($(APP_NAME),)
+$(error APP_NAME is not set)
+endif
+endif
+
+# Check required variables are set for target tag
+ifeq ($(MAKECMDGOALS), tag)
+ifeq ($(APP_NAME),)
+$(error APP_NAME is not set)
+endif
+ifeq ($(DOCKER_USERNAME),)
+$(error DOCKER_USERNAME is not set)
+endif
+endif
+
+# Check required variables are set for target login
+ifeq ($(MAKECMDGOALS), login)
+ifeq ($(DOCKER_PASSWORD),)
+$(error DOCKER_PASSWORD is not set)
+endif
+ifeq ($(DOCKER_USERNAME),)
+$(error DOCKER_USERNAME is not set)
+endif
+endif
+
+# Check required variables are set for target publish
+ifeq ($(MAKECMDGOALS), publish)
+ifeq ($(APP_NAME),)
+$(error APP_NAME is not set)
+endif
+endif
+
+# Set defaults for optional variables
+ifneq ($(DOCKER_SERVER),)
+DOCKER_SERVER := $(DOCKER_SERVER)/
+DOCKER_SERVER_REF = $(DOCKER_SERVER)
+else
+DOCKER_SERVER_REF = <not set>
+endif
+
+DOCKER_USERNAME ?= <not set>
+DOCKER_REPO ?= $(DOCKER_USERNAME)
+TAG ?= $(shell git rev-parse --verify HEAD)
+PUBLISH ?= false
+PATH_PREFIX = opencga-app/app/scripts/docker/$(APP_NAME)
+BRANCH = $(shell git branch | grep \* | cut -d ' ' -f2)
+COMMIT = $(shell git rev-parse --verify HEAD)
+
+# Print useful metadata about the build
+.PHONY: metadata
+metadata:
+	@echo
+	@echo "---------------------"
+	@echo "Metadata"
+	@echo "---------------------"
+	@echo "Branch:          $(BRANCH)"
+	@echo "Commit:          $(COMMIT)"
+	@echo "Docker Tag:      $(TAG)"
+	@echo "Docker Repo:     $(DOCKER_REPO)"
+	@echo "Docker Username: $(DOCKER_USERNAME)"
+	@echo "Docker Server:   $(DOCKER_SERVER_REF)"
+	@echo "Publish:         $(PUBLISH)"
+	@echo "---------------------"
+	@echo
+
+# Builds and tags the container
+.PHONY: build
+build:
+	docker build $(DOCKER_BUILD_ARGS) -t $(APP_NAME) -f $(PATH_PREFIX)/Dockerfile .
+
+# Tags a docker image with git commit SHA
+.PHONY: tag
+tag:
+	docker tag $(APP_NAME) $(DOCKER_SERVER)$(DOCKER_REPO)/$(APP_NAME):$(TAG)
```

Comment:
@joni Just for naming consistency, "DOCKER_REPO" is called organisation "opencb" in DockerHub , we should rename this to DOCKER_ORGANIZATION.

Reply:
Here App name is basically Repository name. 
![image](https://user-images.githubusercontent.com/22955620/51488742-47506480-1d9e-11e9-933e-1f8bfe3a8dcd.png)

Reply:
@wbari thanks, yeh I was going with Azure terminology originally but it makes more sense to move to DH if that's your expected registry. I've updated this now.